src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2NameIdentifier.cs

```diff
@@ -104,7 +104,7 @@ public string Value
             get { return _value; }
             set
             {
-                if (string.IsNullOrEmpty(value))
+                if (string.IsNullOrEmpty(value)) //NOTE: We can remove this check and let our issuer validator handle this.
```

Comment:
With regards to the note, `Saml2NameIdentifier` is used by the issuer, but that is not the only instance.
Unfortunately, it looks like we will need to look more into this as part of the **reading** of the SAML2 token.

I'll add a separate issue to refactor the token reading like we did for JWT.

Reply:
This is related to this issue https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/2905 as well if you'd like to include it into a new issue.